pkg/dogstatsd/replay/writer.go

```diff
@@ -246,6 +246,8 @@ func (tc *TrafficCaptureWriter) Capture(l string, d time.Duration, compressed bo
 		tc.StopCapture()
 	}()
 
+	// copy the instance as `tc.shutdown` can be set at nil
+	shutdown := tc.shutdown
```

Comment:
If this is just selected from, then it shouldn't be a problem: https://go.dev/play/p/JBXCbLNxE4l

I think only the last hunk in this PR (setting tc.shutdown to nil) is required.

Reply:
Sending a message to a channel is thread safe but assigning a channel to nil and reading a message from this channel is not thread safe.
For example, the golang option `--race` reports a data race when running the following code https://go.dev/play/p/gHmyz2jKmxl.